Improve Applications Test

# Story

As an FMN developer,
I want that the test suite is robust to harmless changes outside of FMN,
so that my work isn’t hindered by unnecessary CI failures.

# Acceptance Criteria

The `tests/api/handlers/test_misc.py::TestMisc::test_get_applications` test case …

- [x] … doesn’t fail if new (unknown) messaging schemas for applications are available and installed.
- [x] … verifies application names case-insensitively.

# Background

With new messaging schemas, the list of applications changes, leading to failures like this:

```
=================================== FAILURES ===================================
________________________ TestMisc.test_get_applications ________________________
[gw1] linux -- Python 3.11.2 /__w/fmn/fmn/.tox/py311/bin/python

self = <tests.api.handlers.test_misc.TestMisc object at 0x7f6c8dc5dd50>
client = <starlette.testclient.TestClient object at 0x7f6c8cc5cf90>

    def test_get_applications(self, client):
        response = client.get(f"{self.path}/applications")
    
        assert response.status_code == status.HTTP_200_OK
    
        result = response.json()
        assert isinstance(result, list)
        assert all(isinstance(item, str) for item in result)
        # Verify list is sorted and items are unique
        assert "base" not in result
>       assert result == [
            "anitya",
            "ansible",
            "bodhi",
            "ci_messages",
            "copr",
            "discourse",
            "distgit",
            "elections",
            "FAS",
            "fedocal",
            "FMN",
            "hotness",
            "Koji",
            "mdapi",
            "nuancier",
            "pagure",
            "planet",
        ]
E       AssertionError: assert ['anitya', 'a..., 'Copr', ...] == ['anitya', 'a...scourse', ...]
E         At index 5 diff: 'Copr' != 'discourse'
E         Left contains one more item: 'planet'
E         Use -v to get more diff

tests/api/handlers/test_misc.py:26: AssertionError
```

--- a/tests/api/handlers/test_misc.py
+++ b/tests/api/handlers/test_misc.py
@@ -14,34 +14,46 @@ from .base import BaseTestAPIV1Handler
 
 class TestMisc(BaseTestAPIV1Handler):
     def test_get_applications(self, client):
+        EXPECTED_APPS = {
+            appname.lower()
+            for appname in (
+                "anitya",
+                "ansible",
+                "bodhi",
+                "ci_messages",
+                "copr",
+                "discourse",
+                "distgit",
+                "elections",
+                "FAS",
+                "fedocal",
+                "FMN",
+                "hotness",
+                "Koji",
+                "mdapi",
+                "nuancier",
+                "pagure",
+                "planet",
+            )
+        }
+
         response = client.get(f"{self.path}/applications")
 
         assert response.status_code == status.HTTP_200_OK
 
         result = response.json()
+
         assert isinstance(result, list)
         assert all(isinstance(item, str) for item in result)
-        # Verify list is sorted and items are unique
         assert "base" not in result
-        assert result == [
-            "anitya",
-            "ansible",
-            "bodhi",
-            "ci_messages",
-            "copr",
-            "discourse",
-            "distgit",
-            "elections",
-            "FAS",
-            "fedocal",
-            "FMN",
-            "hotness",
-            "Koji",
-            "mdapi",
-            "nuancier",
-            "pagure",
-            "planet",
-        ]
+
+        result_lower = [appname.lower() for appname in result]
+
+        # Verify list is sorted and items are unique
+        assert result_lower == sorted(set(result_lower))
+
+        # Verify all expected apps are present
+        assert EXPECTED_APPS <= set(result_lower)
 
     @staticmethod
     def mock_distgit_owned_projects(respx_mocker, settings, ownertype):

